pkg/operator/ceph/file/subvolumegroup/controller.go

```diff
@@ -0,0 +1,281 @@
+		}
+
+		// Remove finalizer
+		err = opcontroller.RemoveFinalizer(r.opManagerContext, r.client, cephFilesystemSubVolumeGroup)
+		if err != nil {
+			return reconcile.Result{}, errors.Wrap(err, "failed to remove finalizer")
+		}
+
+		// Return and do not requeue. Successful deletion.
+		return reconcile.Result{}, nil
+	}
+
+	// Build the NamespacedName to fetch the Filesystem and make sure it exists, if not we cannot
+	// create the subvolumegroup
+	cephFilesystem := &cephv1.CephFilesystem{}
+	cephFilesystemNamespacedName := types.NamespacedName{Name: cephFilesystemSubVolumeGroup.Spec.FilesystemName, Namespace: request.Namespace}
+	err = r.client.Get(r.opManagerContext, cephFilesystemNamespacedName, cephFilesystem)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			return reconcile.Result{}, errors.Wrapf(err, "failed to fetch ceph filesystem %q, cannot create subvolumegroup %q", cephFilesystemSubVolumeGroup.Spec.FilesystemName, cephFilesystemSubVolumeGroup.Name)
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, errors.Wrap(err, "failed to get cephFilesystemSubVolumeGroup")
+	}
+
+	// If the CephFilesystem is not ready to accept commands, we should wait for it to be ready
+	if cephFilesystem.Status.Phase != cephv1.ConditionReady {
+		// We know the CR is present so it should a matter of second for it to become ready
+		return reconcile.Result{Requeue: true, RequeueAfter: 10 * time.Second}, errors.Wrapf(err, "failed to fetch ceph filesystem %q, cannot create subvolumegroup %q", cephFilesystemSubVolumeGroup.Spec.FilesystemName, cephFilesystemSubVolumeGroup.Name)
+	}
+
+	// Create or Update ceph filesystem subvolume group
+	err = r.createOrUpdateSubVolumeGroup(cephFilesystemSubVolumeGroup)
+	if err != nil {
+		if strings.Contains(err.Error(), opcontroller.UninitializedCephConfigError) {
+			logger.Info(opcontroller.OperatorNotInitializedMessage)
+			return opcontroller.WaitForRequeueIfOperatorNotInitialized, nil
+		}
+		r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionFailure)
+		return reconcile.Result{}, errors.Wrapf(err, "failed to create or update ceph filesystem subvolume group %q", cephFilesystemSubVolumeGroup.Name)
+	}
+
+	// Success! Let's update the status
+	r.updateStatus(r.client, request.NamespacedName, cephv1.ConditionReady)
+
+	// Return and do not requeue
+	logger.Debug("done reconciling")
+	return reconcile.Result{}, nil
+}
+
+// Create the ceph filesystem subvolume group
+func (r *ReconcileCephFilesystemSubVolumeGroup) createOrUpdateSubVolumeGroup(cephFilesystemSubVolumeGroup *cephv1.CephFilesystemSubVolumeGroup) error {
+	logger.Infof("creating ceph filesystem subvolume group %s in namespace %s", cephFilesystemSubVolumeGroup.Name, cephFilesystemSubVolumeGroup.Namespace)
+
+	// Check if ceph filesystem subvolume group was created manually, create if necessary or update caps and create secret
```

Comment:
updating the caps and creating the secret is not implemented yet, right? Need to change that comment to a TODO?